index.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+var RtmClient = require('@slack/client').RtmClient;
+var RTM_EVENTS = require('@slack/client').RTM_EVENTS;
+var CLIENT_EVENTS = require('@slack/client').CLIENT_EVENTS;
+var schedule = require('node-schedule');
+
+var { util } = require('./app');
+
+
+const botToken = process.env.SLACK_BOT_TOKEN || '';
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+
+if (!botToken || !darkskyToken) {
+    console.error('SLACK_BOT_TOKEN and DARKSKY_TOKEN must be passed as environment variables!');
+    process.exit(1);
+}
+
+var rtm = new RtmClient(botToken);
+
+let botID;
+let botChannel;
+
+rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, (rtmStartData) => {
+    // XXX the private surry-channel doesn't show up in the rtmStartData even though my
+    // bot was invited to the channel :(
+
+    // looks like self.id should always be present - https://api.slack.com/methods/rtm.start
+    botID = rtmStartData.self.id.toLowerCase();
+    console.log(`Logged in as ${rtmStartData.self.name}(${botID}) of team ${rtmStartData.team.name}, but not yet connected to a channel`);
+});
+
+let rtmConnectionOpened = false;
+
+rtm.on(RTM_EVENTS.MESSAGE, function handleRtmMessage(message) {
```

Comment:
I'd also want to break these event handlers up more given more time. They're quite long right now and the code inside can't be reused.